bundles/org.openhab.binding.solarwatt/src/main/java/org/openhab/binding/solarwatt/internal/handler/EnergyManagerHandler.java

```diff
@@ -0,0 +1,403 @@
+     * event after loading all devices.
+     */
+    @Override
+    public void initialize() {
+        this.logger.debug("{} initialize", this);
+        @Nullable
+        SolarwattConfiguration localConfig = this.getConfigAs(SolarwattConfiguration.class);
+        this.initRefresh(localConfig);
+        this.initDeviceCache(localConfig);
+    }
+
+    private void initDeviceCache(SolarwattConfiguration localConfig) {
+        if (localConfig.hostname.isEmpty()) {
+            this.updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Hostname is not set");
+        } else {
+            this.updateStatus(ThingStatus.ONLINE, ThingStatusDetail.CONFIGURATION_PENDING,
+                    "Waiting to retrieve devices.");
+            this.connector.setConfiguration(localConfig);
+
+            this.devicesCache = new ExpiringCache<>(Duration.of(localConfig.refresh, ChronoUnit.SECONDS),
+                    this::refreshDevices);
+
+            @Nullable
+            ExpiringCache<Map<String, Device>> localDevicesCache = this.devicesCache;
+            if (localDevicesCache != null) {
+                // trigger initial load
+                this.scheduler.execute(localDevicesCache::getValue);
+            }
+        }
+    }
+
+    /**
+     * Stop the refresh job and remove devices.
+     */
+    @Override
+    public void dispose() {
+        this.logger.debug("{} dispose", this);
+
+        this.childHandlers.clear();
+
+        @Nullable
+        ScheduledFuture<?> localRefreshJob = this.refreshJob;
+        if (localRefreshJob != null && !localRefreshJob.isCancelled()) {
+            localRefreshJob.cancel(true);
+            this.refreshJob = null;
+        }
+
+        this.devicesCache = null;
+    }
+
+    @Override
+    public void childHandlerInitialized(ThingHandler childHandler, Thing childThing) {
+        String childGuid = childThing.getProperties().get(THING_PROPERTIES_GUID);
+        if (childGuid != null) {
+            this.childHandlers.put(childGuid, childHandler);
+            this.logger.debug("Added handler for thing {}", childThing.getProperties().get(THING_PROPERTIES_GUID));
+        }
+    }
+
+    @Override
+    public void childHandlerDisposed(ThingHandler childHandler, Thing childThing) {
+        if (childThing.getProperties().get(THING_PROPERTIES_GUID) != null) {
+            this.childHandlers.remove(childThing.getProperties().get(THING_PROPERTIES_GUID));
+            this.logger.debug("Removed handler for thing {}", childThing.getProperties().get(THING_PROPERTIES_GUID));
+        }
+    }
+
+    private synchronized void initRefresh(SolarwattConfiguration localConfig) {
+        @Nullable
+        ScheduledFuture<?> localRefreshJob = this.refreshJob;
+        if (localRefreshJob == null || localRefreshJob.isCancelled()) {
+            this.logger.info("Setting Energymanager refreshInterval to '{}' seconds", localConfig.refresh);
```

Comment:
info level should be used very rarely, this looks best suited to debug level especially if the user just changed the refresh interval they should already know it has changed.

https://www.openhab.org/docs/developer/guidelines.html#f-logging

Reply:
Reduced loglevels.